easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -247,8 +257,18 @@ private static void runAnnotatedMethods(Object object, int requestCode) {
 
     private static void checkCallingObjectSuitability(Object object) {
         // Make sure Object is an Activity or Fragment
-        if (!((object instanceof Fragment) || (object instanceof Activity))) {
-            throw new IllegalArgumentException("Caller must be an Activity or a Fragment.");
+        boolean isActivity = object instanceof Activity;
+        boolean isSupportFragment = object instanceof Fragment;
+        boolean isAppFragment = object instanceof android.app.Fragment;
+
+        if (!(isSupportFragment || isActivity ||
+                (isAppFragment && android.os.Build.VERSION.SDK_INT >= 23))) {
+            if (isAppFragment) {
+                throw new IllegalArgumentException(
```

Comment:
I have run into at least [one case](https://code.google.com/p/android/issues/detail?id=191800) where checkSelfPermission did not return true for something I had in my Manifest.  For this reason I'd like to short-circuit the method on lower API levels and let the system deal with permissions.

As for your second point, you're entirely correct.  Forgot about that.